pkg/domain/infra/abi/play.go

```diff
@@ -235,6 +237,18 @@ func (ic *ContainerEngine) PlayKube(ctx context.Context, body io.Reader, options
 			}
 			notifyProxies = append(notifyProxies, proxies...)
 
+			if len(r.Pods) == 1 {
+				pod, err := ic.Libpod.GetPod(r.Pods[0].ID)
+				if err != nil {
+					return nil, err
+				}
+
+				mountLabel, err = pod.MountLabel()
+				if err != nil {
+					return nil, err
+				}
+			}
+
```

Comment:
I don't understand this, looking more into how this works we first sort the kinds in `sortKubeKinds()` so that pods always come last. This means that the `PersistentVolumeClaim` case is always executed before you get here.
Thus the mountlabel will always be empty.

As I read the code you handle it already correctly in `playKubePod()`, so I suggest you remove this part here and from `playKubePVC()`.

Reply:
This comment is still open, please remove this unneeded code or explain what it does.

Reply:
If there is a specified pod and it has a mount label it will be used. This will allow Kubernetes.YAML to specify a file label and have any builtin volumes match the specified label.

Reply:
As described above the sorting makes sure that this never happens and AFAICT the sorting has to happen because volumes must be created before containers otherwise a container could not use them.

Reply:
Well we have to find a way to fix it by parsing the pod spec first before creating the volume, so it can take into effect the specified mountLabel.

Reply:
I want to add the ability to specify the mountLabel for a volume in quadlet, to handle .Volume files, perhaps we can do something there.  Might need to add flag `podman volume create --mount-type`

Reply:
Or specify it as an podman volume create opt mountType=qm_container_file_t ...